astropy/io/fits/convenience.py

```diff
@@ -188,6 +188,7 @@ def getdata(filename, *args, **kwargs):
     data = hdu.data
     if data is None and extidx == 0:
         try:
+            hdulist, extidx = _getext(filename, mode, ext=1, *args, **kwargs)
```

Comment:
I am against hardcoding `ext=1` here. There are all sorts of data formats out there, where the data is not always in Extension 1. This is bound to break someone else's codes.